src/components/authentication/Login.js

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable react/forbid-prop-types */
+/* eslint-disable max-len */
+/* eslint-disable prefer-const */
+/* eslint-disable consistent-return */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-console */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import {
+  Container, Form, Col, Button, Image, Row,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { loginAction } from '../../redux/actions/login';
+import BackgroundImage from '../../assets/backg.jpg';
+import fbImage from '../../assets/f.png';
+import gogleImage from '../../assets/g.png';
+import ErrorAlert from '../sharedComponents/Alert';
+import NavBarComponent from '../sharedComponents/NavbarComponent';
+import Footer from '../sharedComponents/Footer';
+import '../../scss/login.scss';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      loading: false,
+      invalidPasswordFeedback: '',
+      invalidEmailFeedback: '',
+      emailIsValid: false,
+      passwordIsValid: false,
+      disableBtn: true,
+      visible: false,
+      errorMessage: '',
+      textColor: '',
+    };
+  }
+
+  handleChange = (evt) => {
+    this.setState({ passwordIsValid: false, emailIsValid: false });
+    this.setState({ [evt.target.type]: evt.target.value });
+    const { password, email } = this.state;
+    const passwordRegex = /(?=.*\d)(?=.*[A-Z])(?=.*[a-z])(?=.*\W).{7,30}/;
+    const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    const passValidation = passwordRegex.test(password);
+    const emailValidation = emailRegex.test(email);
+    if (!emailValidation) {
+      return this.setState({ invalidEmailFeedback: 'Email must be valid' });
+    }
+    this.setState({ invalidEmailFeedback: '' });
+    if (!passValidation) {
+      return this.setState({ invalidPasswordFeedback: 'Minimum 8 characters, a special charater, number, lower and upper case letters' });
+    }
+    this.setState({ invalidPasswordFeedback: '' });
+    this.setState({
+      emailIsValid: true, passwordIsValid: true, disableBtn: false,
+    });
```

Comment:
Is there any particular reason why you're setting a state directly without using actions and reducers?

Reply:
There isn't any special reason. Should I call an action when all values entered are valid?

Reply:
I guess since I want to provide direct feedback depending on the user inputs, I don't need other actions to update my states.